### PR TITLE
introduce http2

### PR DESCRIPTION
**Pull request by jimpil**
This PR shows a migration path away from `clj-http`. Instead, it uses hato, which wraps the java 11 `HttpClient` (per #118). 

### What I have done

- Created http2.clj mirroring, or at least compatible, with `http.clj` (in its public api, minus the deprecated stuff). In practice, the new api **encourages** being explicit about what server you're talking to (first argument), and what http-options you want to use (last argument). The globally mutable convenience of `connect!` still exists, but is backed behind an atom.
- Switched the `http-api-test` require to `http2` (to make sure the tests pass).
- Added `docker-compose.yml` file (lowers the barrier to contributing)

###  What needs doing (post merge)

- Add licence details at the top of `http2.clj`
- Rename it to `http.clj`, replacing the old/existing one
- Fix the require clause in `http-api-test` 

Let me know if you're happy with how the new namespace looks...It's a little bit longer (even though it has less Vars), but reads nicer/feels simpler, and as I said above, encourages being explicit. Finally, it should be faster, and will take advantage of virtual-threads if available.

### CODE REVIEW

File: src/clojure/langohr/core.clj [hunk 1]
```diff
@@ -244,44 +244,40 @@
 (defn exception-handler
   [{:keys [handle-connection-exception-fn
            handle-return-listener-exception-fn
-           handle-flow-listener-exception-fn
+           ;handle-flow-listener-exception-fn
```
Comment (jimpil) [inline src/clojure/langohr/core.clj:247]: I took the liberty of converting this to `reify` and in the process the compiler complained that `Can't define method not in interfaces: handleFlowListenerException`. I had a look at both `ExceptionListener` (the interface `reify` implements), and `ForgivingExceptionHandler` (the impl that `proxy` was inheriting from), and I can't find that method anywhere - so i removed support for it...

Comment (michaelklishin) [reply src/clojure/langohr/core.clj:247]: Yup, that must have been removed from the Java client a couple of years ago. Langohr was originally based on the interface that was in place around 2011-2013.

**Comment by michaelklishin** [comment]
@jimpil this looks good to me. Thank you!

**Comment by jimpil** [comment]
Cool - do you need anything else from me here?

**Comment by michaelklishin** [comment]
@jimpil resolve the conflicts :) Also, if we are to add a new module, I assume we don't have to bump the version to 6.x? Do you agree?

**Comment by jimpil** [comment]
Conflicts resolved :)
Now, I'm not sure what you mean by ` if we are to add a new namespace ...`. The `http2` namespace is supposed to replace the (existing) `http` one, which allows dropping the `clj-http` dependency. 

Now, regarding bumping major version...If you ask me, I think it's worth doing for two reasons:

1.  First of all, strictly speaking, there are a breaking change in `http2.clj` (dynamic Var(s) became an atom, and deprecated fns were removed). I don't think that will actually affect anyone, but you never know. In addition, the whole philosophy/character of the namespace is somewhat different than before (e.g. you couldn't even do certain things w/o having called `connect!`).
2. `exception-handler` now takes a map of 8 fns maximum (rather than 9). You might say that this is not by any means breaking (extra keys will just be ignored after all), but it is certainly confusing to accept args that do nothing. In any case, this is far less important than the point above.

To summarise...Even you merge my stuff tonight, the thing is not releasable yet. `http.clj` needs to be deleted, `http2.clj` renamed (+ license added) & `http-api-test` require clause fixed. Also, if you happen to have some sort stress-tests or something, now would be the time to run them. Finally, I do recommend a major version bump, but ultimately this is up to you.

**Comment by michaelklishin** [comment]
I will bump the version to a 6.0 preview and you are welcome to continue this work. Thank you!

**Comment by michaelklishin** [comment]
But first, let's rename the branch to `main`, as we did with nearly all other RabbitMQ-related projects.

**Comment by jimpil** [comment]
Hello, I was just wondering what is happening with this work...The PR has been merged, but none of the post-merge actions have been done. Is there a blocker? Do you need help?

**Comment by michaelklishin** [comment]
This has shipped in [`5.5.0`](https://clojars.org/com.novemberain/langohr). Thanks again.
